public/js/grid/react_util.js

```diff
@@ -0,0 +1,115 @@
+var CoursePanel = React.createClass({
+    render: function() {
+        return (
+            <ul className="trapScroll-enabled" id="course-select">
+                <li id="clear-all">
+                    <h3>Clear All</h3>
+                </li>
+            </ul>
+        );
+    }
+});
+
+var SearchPanel = React.createClass({
+    render: function() {
+        return (
+            <div>
+                <div id="filter-container">
+                    <form onsubmit="return false;">
+                        <input id="course-filter" className="form-control" placeholder="Enter a course!" autoComplete="off" type="text" />
+                    </form>
+                </div>
+                <div id="search-container">
+                    <CourseList />
+                </div>
+            </div>
+        );
+    }
+});
+
+var CourseList = React.createClass({
+    getInitialState: function() {
+        return {
+            courses: [],
+            searchList: []
+        };
+    },
+
+    componentDidMount: function() {
+        $.ajax({
+            url: "all-courses",
+            dataType: "text",
+            async: false,
```

Comment:
I don't know, to be honest. Blindly copied this code from the original Javascript file.